src/test/java/net/dv8tion/jda/CacheFlagRequirementsTest.java

```diff
@@ -0,0 +1,176 @@
+        put(PermissionOverrideDeleteEvent.class, EnumSet.of(CacheFlag.MEMBER_OVERRIDES));
+        put(GenericPermissionOverrideEvent.class, EnumSet.of(CacheFlag.MEMBER_OVERRIDES));
+    }};
+
+    private static Reflections events;
+    private static Map<Class<GenericEvent>, EnumSet<CacheFlag>> flagsByClass;
+
+    @BeforeAll
+    static void setup() throws IOException, ClassNotFoundException
+    {
+        final List<CompilationUnit> compilationUnits = parseEventCompilationUnits();
+
+        flagsByClass = getEnumEntriesByClass(CacheFlag.class, compilationUnits);
+
+        events = new Reflections("net.dv8tion.jda.api.events");
+    }
+
+    @Nonnull
+    public static List<CompilationUnit> parseEventCompilationUnits() throws IOException
+    {
+        final SourceRoot root = new SourceRoot(Paths.get("src", "main", "java"));
+        final List<ParseResult<CompilationUnit>> parseResults = root.tryToParse(Event.class.getPackage().getName());
+        return parseResults.stream()
+                .filter(p ->
+                {
+                    if (!p.getProblems().isEmpty())
+                        throw new RuntimeException("Problems when parsing were encountered:\n" + p.getProblems());
+                    else if (!p.getResult().isPresent())
+                        throw new AssertionError("No result was present but no problems were either");
+
+                    return p.getResult().isPresent();
+                })
+                .map(r -> r.getResult().get())
+                .collect(Collectors.toList());
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <E extends Enum<E>> Map<Class<GenericEvent>, EnumSet<E>> getEnumEntriesByClass(Class<E> enumType, List<CompilationUnit> compilationUnits) throws ClassNotFoundException
+    {
+        final Pattern enumEntryReferencePattern = Pattern.compile(Pattern.quote(enumType.getSimpleName()) + "#(\\w+)");
+        final Map<Class<GenericEvent>, EnumSet<E>> enumEntriesByClass = new HashMap<>();
+
+        // Read documented cache flags of scanned classes
+        for (TypeDeclaration<?> typeDeclaration : compilationUnits.stream().flatMap(c -> c.findAll(TypeDeclaration.class).stream()).collect(Collectors.toList()))
+        {
+            final String qualifiedEventName = typeDeclaration.getFullyQualifiedName().orElseThrow(AssertionError::new);
+
+            final JavadocDescription description = typeDeclaration
+                    .getJavadoc()
+                    .map(Javadoc::getDescription)
+                    .orElse(null);
+            if (description == null)
+            {
+                LOGGER.warn("Undocumented class at {}", qualifiedEventName);
+                continue;
+            }
+
+            final List<String> links = getLinks(description);
+            final EnumSet<E> expectedFlags = mapLinks(enumType, links, enumEntryReferencePattern);
+            enumEntriesByClass.put((Class<GenericEvent>) Class.forName(qualifiedEventName), expectedFlags);
```

Comment:
Thoughts on using annotations instead of parsing javadocs? I think it would be a lot easier and clearer if we just added a `@RequiredCacheFlag(CacheFlag.EMOJI)` to the class.